lib/src/rules/avoid_redundant_argument_values.dart

```diff
@@ -71,10 +71,11 @@ class _Visitor extends SimpleAstVisitor {
     for (var i = arguments.length - 1; i >= 0; --i) {
       var arg = arguments[i];
       var param = arg.staticParameterElement;
-      if (param == null || param.hasRequired || param.isRequiredNamed) {
+      if (param == null ||
+          param.hasRequired ||
+          param.isRequiredNamed ||
+          param.isRequiredPositional) {
```

Comment:
We should have an `isRequired` getter on `ParameterElement`, but in the meantime you could use `!param.isOptional`.

Reply:
https://github.com/dart-lang/linter/pull/3086